run_linux.go

```diff
@@ -70,32 +70,29 @@ func setChildProcess() error {
 
 func (b *Builder) cdiSetupDevicesInSpec(deviceSpecs []string, configDir string, spec *specs.Spec) ([]string, error) {
 	leftoverDevices := deviceSpecs
-	registry, err := cdi.NewCache()
+	var configDirs []string
+	defConfig, err := config.Default()
 	if err != nil {
-		return nil, fmt.Errorf("creating CDI registry: %w", err)
+		return nil, fmt.Errorf("failed to get container config: %w", err)
 	}
-	var configDirs []string
+	// The CDI cache prioritizes entries from directories that are later in
+	// the list of ones it scans, so start with the library default, add
+	// our general config, then values passed to us through APIs, in that
+	// order.
+	configDirs = append(slices.Clone(defConfig.Engine.CdiSpecDirs.Get()), configDirs...)
```

Comment:
actually the append here doesn't do anything as configDirs is always empty here. I think this could be simple assignment. `configDirs = slices.Clone(defConfig.Engine.CdiSpecDirs.Get())`

Reply:
Right, and we're not using the library default, either, so I need to correct that comment.